ui/state.go

```diff
@@ -54,3 +57,27 @@ func (win *Window) reloadTabs() {
 	}
 	win.tabs.SetTabs(win.outputs.tabs)
 }
+
+func updateTransactionsWalletState(win *Window, transactions []wallet.TransactionInfo) {
+	win.combined.transactions = nil
+	for i := 0; i < len(transactions); i++ {
+		var transaction struct {
+			iconStatus    *decredmaterial.Icon
+			iconDirection *decredmaterial.Icon
+			info          *wallet.TransactionInfo
+		}
+		transaction.info = &transactions[i]
+		transaction.iconStatus, _ = decredmaterial.NewIcon(icons.ToggleRadioButtonUnchecked)
+		if transactions[i].Status == "Confirmed" {
+			transaction.iconStatus, _ = decredmaterial.NewIcon(icons.ActionCheckCircle)
+			transaction.iconStatus.Color = win.theme.Color.Success
+		}
+		switch transactions[i].Direction {
+		case 0:
+			transaction.iconDirection, _ = decredmaterial.NewIcon(icons.ContentRemove)
+		case 1, 2:
+			transaction.iconDirection, _ = decredmaterial.NewIcon(icons.ContentAdd)
+		}
+		win.combined.transactions = append(win.combined.transactions, transaction)
+	}
+}
```

Comment:
This shouldn't be necessary, on receiving a `TransactionInfo` slice the window should add it to its `walletTransactions`